src/core/server/elasticsearch/scoped_cluster_client.ts

```diff
@@ -72,10 +72,9 @@ export class ScopedClusterClient {
     clientParams: Record<string, unknown> = {},
     options?: CallAPIOptions
   ) {
-    if (this.headers !== undefined) {
-      clientParams.headers = this.headers;
+    if (this.headers || clientParams.headers) {
+      clientParams.headers = Object.assign({}, clientParams.headers, this.headers);
```

Comment:
The new `elasticsearch-js` client has the opposite behavior that you define here, where the request-specific headers override any headers already set on the transport (for all requests): https://github.com/elastic/elasticsearch-js/blob/master/lib/Transport.js#L121

Is there a specific reason why you wanted it to work the other way?

If the behavior you have here _is_ the right way to move forward, then I think we should throw an error if you attempt to make a request with a header that conflicts with a client-wide header rather than just ignoring the value.

Reply:
My line of through was that the previous implementation totally ignored user sent headers. So this attempt is safer as it still favors the predefined headers.

I'd vote for simplicity rather than value checking and throwing errors. Just a consistent behavior with a simple implementation. Moving in that direction means changing the code to what is presented in the transport.

```suggestion
      clientParams.headers = Object.assign({}, this.headers, clientParams.headers);
```